File: old_src_backup/frontend/types/todo.ts
// Task interface definition
export interface Task {
  id: string;
  title: string;
  description?: string;
  is_completed: boolean;
  created_at: string; // ISO string format
  updated_at: string; // ISO string format
  due_date?: string; // Optional due date in ISO string format
  priority?: 'low' | 'medium' | 'high'; // Optional priority level
  category?: string; // Optional category
  owner_id?: string; // Optional owner ID
}

// Interface for creating a new task (without id and timestamps)
export interface CreateTaskData {
  title: string;
  description?: string;
  is_completed?: boolean;
  due_date?: string;
  priority?: 'low' | 'medium' | 'high';
  category?: string;
}

// Interface for updating a task (all fields optional)
export interface UpdateTaskData {
  title?: string;
  description?: string;
  is_completed?: boolean;
  due_date?: string;
  priority?: 'low' | 'medium' | 'high';
  category?: string;
}

// Interface for API response when fetching tasks
export interface GetTasksResponse {
  tasks: Task[];
  total: number;
  page: number;
  limit: number;
}

File: old_src_backup/backend/api/schemas/todo.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
import uuid


class TodoBase(BaseModel):
    title: str
    description: Optional[str] = None
    is_completed: bool = False
    priority: int = 2  # 1: low, 2: medium, 3: high, 4: urgent
    project_id: Optional[uuid.UUID] = None


class TodoCreate(TodoBase):
    title: str
    pass


class TodoUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    is_completed: Optional[bool] = None
    priority: Optional[int] = None
    project_id: Optional[uuid.UUID] = None


class TodoResponse(TodoBase):
    id: uuid.UUID
    owner_id: uuid.UUID
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True